### ISSUE TITLE
↔ wrongfully turned into emoji variant

### ISSUE THREAD

**Issue by the-eater** [mastodon/mastodon#26089]
### Steps to reproduce the problem

1. Post ↔

### Expected behaviour

The post contains ↔

### Actual behaviour

The post contains an svg/emoji of ↔

### Detailed description

↔ is being wrongfully turned into it's emoji equivalent, the default presentation of ↔ (aka without variant selector) should be the text form, as defined in the Unicode spec (while Emoji=Yes on this character, since Emoji_Presentation=No, it needs to be shown in it's text form unless a emoji variant modifier is added)

https://util.unicode.org/UnicodeJsps/character.jsp?a=2194&B1=Show
https://unicode.org/reports/tr51/proposed.html#Emoji_Variation_Selector_Notes

This issue might not be limited to only this character, but the emoji handling of Mastodon isn't clear to me, thus I am unable to identify all of them in this case

TL;DR;

`\u2194` should show as text (↔︎)
`\u2194\ufe0f` should show as :left_right_arrow: 

### Specifications

Mastodon v4.1.4+nightly-20230718 (I tested this against Mastodon.social)

**Comment by the-eater** [comment]
et tu github

**Comment by ClearlyClaire** [comment]
This was chosen to be consistent with Twitter's behavior, as this is what most people would expect. Adding VS15 should make it show the non-emojified version.

**Comment by nabijaczleweli** [comment]
You'll probably find that most people would expect that the webui is consistent with (a) their mobile client (tusky renders this correctly), and (b) *what they typed in*. My ass is not adding random extraneous variation selectors to make the render (for which there's no preview, because why would there be – it always shows what you typed in) show what I typed in, because I already typed in what I want.

The webui Emoji Selector does yield
```
↔	(U+2194, UTF-8 e2 86 94 ): LEFT RIGHT ARROW
	(U+FE0F, UTF-8 ef b8 8f ): VARIATION SELECTOR-16
```
so if people want to insert emoji arrows they very much can, but if they want to insert real arrows they... need to type in arrows, then   add in Some Random Modifier (which, prior to this, I didn't know was a thing!), and it's only broken in the webui? come on

**Comment by the-eater** [comment]
I personally can't see how "people that copy or write a ↔︎ expect it to turn into ↔️" would be true, but that's subjective.

And nonetheless that it's not non-conformant to the Unicode spec, Twitter does this at composing, Mastodon does it at rendering

when I write ↔︎ in a Twitter textbox, I get to see a :left_right_arrow: immediately,
when I write ↔︎ in a Mastodon textbox, I just get to see said ↔︎, and only after I post, I can see that it was turned into  :left_right_arrow:

**Comment by dsalt** [comment]
I consider rendering the colour emoji variant regardless of `U+FE0E VARIATION SELECTOR 15` to be a bug.

Using `😆` as an example here (so, basic form `😆︎`):

I note that it's correctly rendered while editing a posting, but otherwise the colour emoji variant is shown. This is misleading, and actually broke a posting of mine earlier: I'd deliberately typed the forced basic form (I have X11 compose sequences for such things).

And no, Tusky *doesn't* render it correctly: I have a test posting which shows both colour emoji and basic forms, but both are rendered as colour emojis.
